lib/macho.rb

```diff
@@ -56,6 +56,6 @@ def self.codesign!(filename)
            "--preserve-metadata=entitlements,requirements,flags,runtime",
            filename)
 
-    raise ModificationError, "#{filename}: signing failed!" unless $CHILD_STATUS.success?
+    $stderr.puts "#{filename}: signing failed!" unless $CHILD_STATUS.success?
```

Comment:
I think raising an exception does make sense here but having a specific, public exception class for code signing would be nice to it can be rescued in Homebrew/brew

Reply:
Yeah, agreed. We could do `CodeSigningError < ModificationError` if that works for you, @mistydemeo?

Reply:
Rebased and changed to that. Agreed we can rescue in Homebrew rather than here.